sdk/core/azure-core/src/cryptography/md5.cpp

```diff
@@ -16,100 +16,71 @@
+namespace {
 
 #if defined(AZ_PLATFORM_WINDOWS)
 
-  namespace _detail {
-    struct AlgorithmProviderInstance
-    {
-      BCRYPT_ALG_HANDLE Handle;
-      std::size_t ContextSize;
-      std::size_t HashLength;
-
-      AlgorithmProviderInstance()
-      {
-        NTSTATUS status = BCryptOpenAlgorithmProvider(&Handle, BCRYPT_MD5_ALGORITHM, nullptr, 0);
-        if (!BCRYPT_SUCCESS(status))
-        {
-          throw std::runtime_error("BCryptOpenAlgorithmProvider failed");
-        }
-        DWORD objectLength = 0;
-        DWORD dataLength = 0;
-        status = BCryptGetProperty(
-            Handle,
-            BCRYPT_OBJECT_LENGTH,
-            reinterpret_cast<PBYTE>(&objectLength),
-            sizeof(objectLength),
-            &dataLength,
-            0);
-        if (!BCRYPT_SUCCESS(status))
-        {
-          throw std::runtime_error("BCryptGetProperty failed");
-        }
-        ContextSize = objectLength;
-        DWORD hashLength = 0;
-        status = BCryptGetProperty(
-            Handle,
-            BCRYPT_HASH_LENGTH,
-            reinterpret_cast<PBYTE>(&hashLength),
-            sizeof(hashLength),
-            &dataLength,
-            0);
-        if (!BCRYPT_SUCCESS(status))
-        {
-          throw std::runtime_error("BCryptGetProperty failed");
-        }
-        HashLength = hashLength;
-      }
-
-      ~AlgorithmProviderInstance() { BCryptCloseAlgorithmProvider(Handle, 0); }
-    };
-
-    struct Md5HashContext
-    {
-      std::string buffer;
-      BCRYPT_HASH_HANDLE hashHandle = nullptr;
-      std::size_t hashLength = 0;
-    };
-  } // namespace _detail
+struct Md5AlgorithmProvider
+{
+  BCRYPT_ALG_HANDLE Handle;
+  std::size_t ContextSize;
+  std::size_t HashLength;
 
-  Md5Hash::Md5Hash()
+  Md5AlgorithmProvider()
   {
-    static _detail::AlgorithmProviderInstance AlgorithmProvider{};
-
-    _detail::Md5HashContext* md5Context = new _detail::Md5HashContext;
-    m_md5Context = md5Context;
-    md5Context->buffer.resize(AlgorithmProvider.ContextSize);
-    md5Context->hashLength = AlgorithmProvider.HashLength;
-
-    NTSTATUS status = BCryptCreateHash(
-        AlgorithmProvider.Handle,
-        &md5Context->hashHandle,
-        reinterpret_cast<PUCHAR>(&md5Context->buffer[0]),
-        static_cast<ULONG>(md5Context->buffer.size()),
-        nullptr,
-        0,
+    NTSTATUS status = BCryptOpenAlgorithmProvider(&Handle, BCRYPT_MD5_ALGORITHM, nullptr, 0);
+    if (!BCRYPT_SUCCESS(status))
+    {
+      throw std::runtime_error("BCryptOpenAlgorithmProvider failed");
+    }
+    DWORD objectLength = 0;
+    DWORD dataLength = 0;
+    status = BCryptGetProperty(
+        Handle,
+        BCRYPT_OBJECT_LENGTH,
+        reinterpret_cast<PBYTE>(&objectLength),
+        sizeof(objectLength),
+        &dataLength,
         0);
     if (!BCRYPT_SUCCESS(status))
     {
-      throw std::runtime_error("BCryptCreateHash failed");
+      throw std::runtime_error("BCryptGetProperty failed");
```

Comment:
As a developer/customer what would I do when I see this error?  I suggest adding in the status code or expanding the status code into text to help diagnose the failure.